From: Jan Kara <jack@suse.cz>
Subject: readahead: drop dead code in page_cache_ra_order()
Date: Tue, 25 Jun 2024 12:18:56 +0200

page_cache_ra_order() scales folio order down so that is fully fits within
readahead window.  Thus the code handling the case where we walked past
the readahead window is a dead code.  Remove it.

Link: https://lkml.kernel.org/r/20240625101909.12234-6-jack@suse.cz
Signed-off-by: Jan Kara <jack@suse.cz>
Reviewed-by: Josef Bacik <josef@toxicpanda.com>
Tested-by: Zhang Peng <zhangpengpeng0808@gmail.com>
Cc: Matthew Wilcox (Oracle) <willy@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/readahead.c |    5 -----
 1 file changed, 5 deletions(-)

--- a/mm/readahead.c~readahead-drop-dead-code-in-page_cache_ra_order
+++ a/mm/readahead.c
@@ -525,11 +525,6 @@ void page_cache_ra_order(struct readahea
 		index += 1UL << order;
 	}
 
-	if (index > limit) {
-		ra->size += index - limit - 1;
-		ra->async_size += index - limit - 1;
-	}
-
 	read_pages(ractl);
 	filemap_invalidate_unlock_shared(mapping);
 	memalloc_nofs_restore(nofs);
_
